packages/target-tools/src/telemetryProvider.js

```diff
@@ -10,14 +20,24 @@ import { DECISIONING_METHOD } from "./enums";
 export function TelemetryProvider(
   executeTelemetriesFunc,
   telemetryEnabled = true,
-  mode = DECISIONING_METHOD.SERVER_SIDE
+  method = DECISIONING_METHOD.SERVER_SIDE
 ) {
   let telemetryEntries = [];
 
+  function getMode(status) {
+    if (method === DECISIONING_METHOD.HYBRID && status === PARTIAL_CONTENT) {
+      return EXECUTION_MODE.ON_DEVICE;
```

Comment:
@dcottingham I have some doubts about this new "execution mode" concept that we're adding, did we have a spec/discussion around this and do we really need it in addition to decisioning method? Why not just add new values to decisioning method, like `HYBRID_PARTIAL` and `HYBRID_FULL`?
Also, is there a corresponding m2 PR adding this new field to Delivery API?